src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/EmbeddedLanguages/VirtualChars/CSharpVirtualCharService.cs

```diff
@@ -145,6 +161,72 @@ private static VirtualCharSequence TryConvertRawStringToVirtualChars(SyntaxToken
             return CreateVirtualCharSequence(tokenText, offset, startIndexInclusive, endIndexExclusive, result);
         }
 
+        private static VirtualCharSequence TryConvertMultiLineRawStringToVirtualChars(SyntaxToken token)
+        {
+            // The containing expression for this token.  This is needed so that we can determine the indentation
+            // whitespace based on the last line of the containing multiline literal.
+            ExpressionSyntax parentExpression = token.Kind() is SyntaxKind.MultiLineRawStringLiteralToken or SyntaxKind.UTF8MultiLineRawStringLiteralToken
+                ? (LiteralExpressionSyntax)token.GetRequiredParent()
+                : (InterpolatedStringExpressionSyntax)token.GetRequiredParent().GetRequiredParent();
+
+            // Whether or not the token contains the delimiters, or if the parent owns them.  This will affect if we
+            // need to skip looking at the start/end line of the token.
+            var hasDelimiters = parentExpression is LiteralExpressionSyntax;
+
+            // if this is the first text content chunk of the multi-line literal.  The first chunk contains the leading
+            // indentation of the line it's on (which thus must be trimmed), while all subsequent chunks do not (because
+            // they start right after some `{...}` interpolation
+            var isFirstChunk =
+                parentExpression is LiteralExpressionSyntax ||
+                (parentExpression is InterpolatedStringExpressionSyntax { Contents: var contents } && contents.First() == token.GetRequiredParent());
+
+            if (parentExpression.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error))
+                return default;
+
+            // Use the parent multi-line expression to determine what whitespace to remove from the start of each line.
+            var parentSourceText = parentExpression.SyntaxTree.GetText();
+            var indentationLength = parentSourceText.Lines.GetLineFromPosition(parentExpression.Span.End).GetFirstNonWhitespaceOffset() ?? 0;
```

Comment:
I was slightly confused for a second because I didn't notice the Span.End - might be good to call out that this is explicitly looking at the indentation of the closing `""";` piece

Reply:
sure :)